File: terceiroPeriodo/aeds-ii/tps/tp-iv/ex03/assets/logStatus.c
#include <stdlib.h>
#include <time.h>
#include "logStatus.h"

LOGSTATUS* new_logstatus(){
	LOGSTATUS* resp = (LOGSTATUS*)calloc(1,sizeof(LOGSTATUS));

	resp->comparacoes = 0;
	resp->inicio = resp->fim = 0;

	return resp;
}

void iniciarContagem(LOGSTATUS *ls){
	ls->inicio = clock();
}
void terminarContagem(LOGSTATUS *ls){
	ls->fim = clock();
}

void comparacao(LOGSTATUS *ls,int i){
	ls->comparacoes += i;
}

double getTime(LOGSTATUS *ls){
	double resp = ((double) (ls->fim - ls->inicio) / CLOCKS_PER_SEC) * 1000.0;
	return resp;
}

File: terceiroPeriodo/aeds-ii/tps/tp-iv/ex07/assets/LogStatus.java
package ex06.assets;

public class LogStatus{
	private Long inicio;
	private Long fim;
	private int comparacoes;

	public LogStatus(){
		comparacoes = 0;
		this.inicio = this.fim = null;
	}

	public void iniciar(){
		this.inicio = System.currentTimeMillis();
	}

	public void parar(){
		this.fim = System.currentTimeMillis();
	}

	public Long tempoAlgoritmo(){
		if(inicio != null && fim != null){
			return fim - inicio;
		}
		return 0L;
	}
	
	public void comparacao(int x){
		comparacoes+=x;
	}

	public int getComparacoes() {
		return comparacoes;
	}
}
